### PR TITLE
Enable editorconfig checker and reduce indentation for a few modules

### PR DESCRIPTION
**Pull request by zliu41**
No code change, purely re-formatting.
Enabled editorconfig checker for some modules (which enforces line length <= 100), and reduced indentation to 2 spaces.

### CODE REVIEW

**Comment by michaelpj** [comment]
> reduced indentation to 2 spaces

https://github.com/input-output-hk/plutus/blob/master/STYLEGUIDE.adoc#basic-formatting

If you want to use 2 spaces we should have the argument and change the style guide first :)

**Comment by michaelpj** [comment]
Then, let's change the style guide too!

File: plutus-core/plutus-ir/src/PlutusIR/Compiler/Definitions.hs [hunk 1]
```diff
@@ -11,25 +10,28 @@
...
-                                              , lookupTerm
-                                              , lookupOrDefineTerm
-                                              , lookupType
-                                              , lookupOrDefineType
-                                              , lookupConstructors
-                                              , lookupDestructor) where
+module PlutusIR.Compiler.Definitions (
+  DefT,
+  MonadDefs (..),
+  TermDefWithStrictness,
```
Comment (effectfully) [inline plutus-core/plutus-ir/src/PlutusIR/Compiler/Definitions.hs:18]: So let's do that here...

File: plutus-core/plutus-ir/src/PlutusIR/Compiler/Definitions.hs [hunk 2]
```diff
@@ -73,157 +75,238 @@ mapDefs :: (a -> b) -> DefMap key a -> DefMap key b
...
+  }
 makeLenses ''DefState
 
-newtype DefT key uni fun ann m a = DefT { unDefT :: StateT (DefState key uni fun ann) m a }
-    deriving newtype (Functor, Applicative, Monad, MonadTrans, MM.MFunctor, MonadError e, MonadReader r, MonadQuote, MonadWriter w)
+newtype DefT key uni fun ann m a = DefT {unDefT :: StateT (DefState key uni fun ann) m a}
+  deriving newtype
+    ( Functor
+    , Applicative
+    , Monad
```
Comment (effectfully) [inline plutus-core/plutus-ir/src/PlutusIR/Compiler/Definitions.hs:92]: ... and that here.

File: plutus-core/plutus-ir/src/PlutusIR/Subst.hs [hunk 3]
```diff
@@ -26,96 +26,111 @@ import Data.Set.Lens (setOf)
...
 funRes :: Lens' (Type tyname uni a) (Type tyname uni a)
 funRes f = \case
-    TyFun a dom cod -> TyFun a dom <$> funRes f cod
-    t               -> f t
+  TyFun a dom cod -> TyFun a dom <$> funRes f cod
+  t               -> f t
 
 -- TODO: these could be Traversals
+
 -- | Get all the term variables in a term.
```
Comment (effectfully) [inline plutus-core/plutus-ir/src/PlutusIR/Subst.hs:134]: Another example.

**Review by effectfully — COMMENTED** [review]
Nice try, but you won't be able to sneakily introduce `ormolu` like that :stuck_out_tongue:

**Comment by zliu41** [comment]
> Nice try, but you won't be able to sneakily introduce ormolu like that stuck_out_tongue

It's fourmolu, much less opinionated than ormolu and very configurable, e.g., doesn't mess with import blocks :)

I don't really care about fourmolu and certainly won't mandate it, and am happy to address your comments. The things I definitely want are (1) two space indentation (2) normalizing at least the type signatures (in particular, `foo ::` rather than `foo\n  ::`, which makes it much easier to find the definition of `foo`) - there's no need to have so many variations of type signatures.

**Pull request merged**